server/render/test/index.test.js

```diff
@@ -0,0 +1,249 @@
+} );
+
+jest.mock( 'state/ui/selectors', () => {
+	return {
+		isSectionIsomorphic: () => mockReturnValues.isSectionIsomorphic,
+	};
+} );
+
+jest.mock( 'config', () => {
+	const fn = () => {};
+	fn.isEnabled = feature_key =>
+		feature_key === 'server-side-rendering' ? mockReturnValues.configServerSideRender : false;
+	return fn;
+} );
+
+const ssrCompatibleContext = {
+	layout: 'hello',
+	user: null,
+	store: {
+		getState: () => {},
+	},
+	lang: 'en',
+};
+
+const ssrEnabledContext = {
+	...ssrCompatibleContext,
+	serverSideRender: true,
+};
+
+const ssrDisabledContext = {
+	...ssrCompatibleContext,
+	serverSideRender: false,
+};
+
+describe( 'shouldServerSideRender', () => {
+	beforeEach( () => remock() );
+
+	test( 'feature-flag server-side-render should enable SSR (default behavior)', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+	} );
+
+	test( 'feature-flag server-side-render should disable SSR', () => {
+		remock( { configServerSideRender: false } );
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( false );
+	} );
+
+	test( 'context.serverSideRender should alter the result', () => {
+		expect( shouldServerSideRender( ssrCompatibleContext ) ).toBe( false ); // due to undefined
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+		expect( shouldServerSideRender( ssrDisabledContext ) ).toBe( false );
+	} );
+
+	test( 'context.layout should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		const SsrEnabledContextWithoutLayout = {
+			...ssrEnabledContext,
+			layout: undefined,
+		};
+		expect( shouldServerSideRender( SsrEnabledContextWithoutLayout ) ).toBe( false );
+	} );
+
+	test( 'context.user should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		const ssrEnabledContextWithUser = {
+			...ssrEnabledContext,
+			user: {
+				name: 'hello-world',
+			},
+		};
+		expect( shouldServerSideRender( ssrEnabledContextWithUser ) ).toBe( false );
+	} );
+
+	test( 'isSectionIsomorphic should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		remock( { isSectionIsomorphic: false } );
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( false );
+	} );
+
+	test( 'isDefaultLocale should alter the result', () => {
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( true );
+
+		remock( { isDefaultLocale: false } );
+		expect( shouldServerSideRender( ssrEnabledContext ) ).toBe( false );
+	} );
+} );
+
+describe( 'setShouldServerSideRender', () => {
+	test( 'when query is empty, then sets context.serverSideRender to TRUE - and calls next()', () => {
+		const next = jest.fn();
+		const contextWithoutQueryKeys = {
+			query: {},
+		};
+
+		setShouldServerSideRender( contextWithoutQueryKeys, next );
+		expect( contextWithoutQueryKeys.serverSideRender ).toBe( true );
+		expect( next.mock.calls.length ).toBe( 1 );
```

Comment:
Can we use the more semantic `expect( next ).toHaveBeenCalledTimes( 1 );` here? See https://jest-bot.github.io/jest/docs/expect.html#tohavebeencalledtimesnumber

Reply:
Nice syntax! All occurrences replaced.